src/frontend/ts/state/chessStore.ts

```diff
@@ -186,6 +193,12 @@ const useChessStore = create<ChessState>((set, get) => ({
 
     set({ incomingChallenges })
   },
+  removeOutgoingChallenge: (who: Ship) => {
+    let outgoingChallenges: Map<Ship, Challenge> = get().outgoingChallenges
+    outgoingChallenges.delete(who)
+
+    set({ outgoingChallenges })
+  },
```

Comment:
Sorry, I should have left a comment about this earlier. The `removeOutgoing` function in `ChessState` that's used in `Challenges` is actually precisely the thing that we're trying to get rid of (as well as the `onSuccess` callbacks in `acceptChallenge` and `declineChallenge`). This should still work as you have written by copying the contents of `removeChallenge` and `removeOutgoingChallenge` into the `receiveChallenge` function (which probably should be renamed to `receiveChallengeUpdate`):
```
...
case Update.ChallengeDeclined: {
  let outgoingChallenges: Map<Ship, Challenge> = get().outgoingChallenges
  outgoingChallenges.delete(data.who)

  set({ outgoingChallenges })
  break
}
case Update.ChallengeResolved: {
  let incomingChallenges: Map<Ship, Challenge> = get().incomingChallenges
  incomingChallenges.delete(data.who)

  set({ incomingChallenges })
  break
}
...
```